runtimes/pythonrt/runner/main.py

```diff
@@ -262,22 +247,26 @@ def ActivityReply(
             log.exception(f"can't decode data: pickle: {err}")
             abort_with_exception(context, grpc.StatusCode.INTERNAL, err)
 
+        if not isinstance(result, Result):
+            context.abort(
+                grpc.StatusCode.INVALID_ARGUMENT, "ActivityReply data not a Result"
+            )
+
         with self.lock:
-            call = self.activity_calls.pop() if self.activity_calls else None
+            call = self.activity_call
+            self.activity_call = None
 
         if call is None:
             context.abort(
                 grpc.StatusCode.INVALID_ARGUMENT, "ActivityReply without pending calls"
             )
 
-        if request.error:
-            call.fut.set_exception(ActivityError(request.error))
-            context.abort(
-                grpc.StatusCode.ABORTED,
-                f"activity error: {request.error}",
-            )
+        if result.error:
+            # TODO: Restore exception traceback
```

Comment:
can you explain?

Reply:
When you pickle an exception, the stored data does not have the traceback associated with the exception (you can't pickle a stack trace). I want to add the traceback back to the exception somehow, though not urgent since it's printed out.